apps/builder/app/builder/features/style-panel/sections/filter/filter.tsx

```diff
@@ -0,0 +1,82 @@
+import { CollapsibleSectionBase } from "~/builder/shared/collapsible-section";
+import type { RenderCategoryProps } from "../../style-sections";
+import { useState } from "react";
+import {
+  SectionTitle,
+  SectionTitleButton,
+  SectionTitleLabel,
+  Tooltip,
+} from "@webstudio-is/design-system";
+import { getDots } from "../../shared/collapsible-section";
+import { PropertyName } from "../../shared/property-name";
+import { PlusIcon } from "@webstudio-is/icons";
+import { type StyleProperty } from "@webstudio-is/css-engine";
+import { getStyleSource } from "../../shared/style-info";
+import { LayersList } from "../../style-layers-list";
+import { FilterLayer } from "./filter-layer";
+import { addLayer } from "../../style-layer-utils";
+import { parseFilter } from "@webstudio-is/css-data";
+
+const property: StyleProperty = "filter";
+const label = "Filter";
+
+export const FilterSection = (props: RenderCategoryProps) => {
+  const { currentStyle, deleteProperty } = props;
+  const [isOpen, setIsOpen] = useState(true);
+  const layerStyleSource = getStyleSource(currentStyle[property]);
+  const filterValue = currentStyle[property]?.value;
+
+  return (
+    <CollapsibleSectionBase
+      fullWidth
+      label={label}
+      isOpen={isOpen}
+      onOpenChange={setIsOpen}
+      trigger={
+        <SectionTitle
+          dots={getDots(currentStyle, [property])}
+          suffix={
+            <Tooltip content={"Add a filter"}>
+              <SectionTitleButton
+                prefix={<PlusIcon />}
+                onClick={() => {
+                  addLayer(
+                    property,
+                    parseFilter("blur(0px)"),
+                    currentStyle,
+                    props.createBatchUpdate
+                  );
+                  setIsOpen(true);
+                }}
+              />
+            </Tooltip>
+          }
+        >
+          <PropertyName
+            title={label}
+            style={currentStyle}
+            properties={[property]}
+            description="Filter effects allow you to apply graphical effects like blurring, color shifting, and more to elements."
+            label={
+              <SectionTitleLabel color={layerStyleSource}>
+                {label}
+              </SectionTitleLabel>
+            }
+            onReset={() => deleteProperty(property)}
+          />
+        </SectionTitle>
+      }
+    >
+      {filterValue?.type === "tuple" && filterValue.value.length > 0 && (
```

Comment:
Am I right that we use tuple type here just because it's joined with space in our style rendering https://github.com/webstudio-is/webstudio-builder/blob/de7f764932477e6702d481e9459c16f5bfd1d62a/packages/css-engine/src/core/to-value.ts#L112
Would be good to comment this internal knowledge. Or probably we need copy of Tuple type with some semantically correct name. Not sure